### PR TITLE
Python. Homework #2. Task #3. Вильданов Эмир

### PR DESCRIPTION
**Pull request by EmirVildanov**

### CODE REVIEW

File: homework/hw2/task3/smart_args.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import functools
+from collections.abc import Callable
+from inspect import signature, getfullargspec
+from copy import deepcopy
+import random
```
Comment (nbirillo) [inline homework/hw2/task3/smart_args.py]: Please, delete unused imports

File: homework/hw2/task3/smart_args.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+def smart_args(func=None, *, positional_args_on: bool = False) -> Callable:
+    """
+    :param func: inner function for decorator
+    :param positional_args_on: if True decorator will also watch through positional arguments
+    :return: inner function with processed Isolated and Evaluated arguments
+    """
+    if func is None:
+        if positional_args_on:
+            return lambda func: smart_args(func, positional_args_on=True)
+        return lambda func: smart_args(func)
```
Comment (nbirillo) [inline homework/hw2/task3/smart_args.py]: It looks a little bit strange. Why did not you use:
```
return lambda func: smart_args(func, positional_args_on=positional_args_on)
```
?

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great, thanks! I have two little comments

File: homework/hw2/task3/smart_args.py (hunk 2 above)
Comment (EmirVildanov) [reply homework/hw2/task3/smart_args.py]: Cause writing `if` statement doesn't require a lot brain work. I fixed it)

**Review by nbirillo — APPROVED** [review]
Great!

**Pull request merged**